scipy/optimize/_differentialevolution.py

```diff
@@ -503,8 +545,13 @@ def __init__(self, func, bounds, args=(),
         self.polish = polish
 
         # set the updating / parallelisation options
-        if updating in ['immediate', 'deferred']:
+        if updating in ['immediate', 'deferred', 'worst', 'plus']:
             self._updating = updating
+        else:
+            raise ValueError(
+                "The updating keyword argument must be one of {'immediate', "
+                "'deferred', 'worst', 'plus'}"
+            )
```

Comment:
```suggestion
        methods = {'immediate', 'deferred', 'worst', 'plus'}
        if updating in methods:
            self._updating = updating
        else:
            message = f"The updating keyword argument must be one of {methods}"
            raise ValueError(message)
```
Mostly for future maintenance.